ccm_web/server/src/main.ts

```diff
@@ -4,47 +4,40 @@ import { urlencoded, json } from 'express'
 import session from 'express-session'
 import morgan from 'morgan'
 import { Sequelize } from 'sequelize-typescript'
-import { ValidationPipe } from '@nestjs/common'
+import { INestApplication, ValidationPipe } from '@nestjs/common'
 import { ConfigService } from '@nestjs/config'
 import { NestFactory } from '@nestjs/core'
 import { NestExpressApplication } from '@nestjs/platform-express'
 import { DocumentBuilder, SwaggerModule } from '@nestjs/swagger'
 
 import { AppModule } from './app.module'
 
-import { Config } from './config'
+import { Config, ServerConfig } from './config'
 import baseLogger from './logger'
 
 const logger = baseLogger.child({ filePath: __filename })
 
-async function bootstrap (): Promise<void> {
-  const isDev = process.env.NODE_ENV !== 'production'
+type PartialServerConfig = Omit<ServerConfig, 'isDev' | 'port'>
 
-  const app = await NestFactory.create<NestExpressApplication>(AppModule, {
-    logger: !isDev ? ['warn', 'error'] : undefined
-  })
-
-  const configService = app.get<ConfigService<Config, true>>(ConfigService)
-  const sequelize = app.get(Sequelize)
-  const serverConfig = configService.get('server', { infer: true })
+export function doAppCoreSetup (app: INestApplication, serverConfig: PartialServerConfig): void {
+  // Logging
   baseLogger.level = serverConfig.logLevel
-
   const stream = { write: (message: string) => { logger.info(message.trim()) } }
   app.use(morgan('combined', { stream: stream }))
 
-  app.set('trust proxy', 1)
+  // Controls size limit of data in payload and URL
+  const payloadSizeLimit = '5mb'
+  app.use(json({ limit: payloadSizeLimit }))
+  app.use(urlencoded({ extended: true, limit: payloadSizeLimit }))
 
+  // Cookies and Sessions
   app.use(cookieParser(serverConfig.cookieSecret))
 
+  const sequelize = app.get(Sequelize)
   const SequelizeStore = ConnectSessionSequelize(session.Store)
```

Comment:
Any reason for change of ordering?

Reply:
Not too much, I guess I figured I'd just get `sequelize` when it was needed, kinda keeps the related code together. I could see either way (early set up, or set up just in time) being okay.